### ISSUE TITLE
New Cop Idea: Explicit operator precedence

### ISSUE THREAD

**Issue by tejasbubane** [rubocop/rubocop#7849]
## Is your feature request related to a problem? Please describe.

This is a request for new cop based on @bbatsov's comment [here](https://github.com/rubocop-hq/rubocop/pull/7847#discussion_r403443811).

## Describe the solution you'd like

The cop should check for explicit parens when operators with varying precedence are used in a statement.

```ruby
# bad
a && b || c
10 + 20 * 3

# good
(a && b) || c
10 + (20 * 3)
```

## Describe alternatives you've considered

NA

## Additional context

We need to decide if the cop should consider all operators or only commonly used ones.

**Comment by bbatsov** [comment]
Well, I have to say I love the idea! 😉

**Comment by rahul404** [comment]
Hi @bbatsov, I have made a PR for this feature. There is one issue I can fix in it.
The test case to fix `a ** b * c / d % e + f - g << h >> i & j | k ^ l` fails. Somehow the cop exercises the `correction` method only once.

**Comment by mvz** [comment]
It would be nice to have this depend somehow on the complexity of the expression. But then again, I think `a ** b * c / d % e + f - g << h >> i & j | k ^ l` would be equally hard to comprehend *with* parentheses.
